servlet-engine/src/main/java/io/micronaut/servlet/engine/DefaultServletHttpRequest.java

```diff
@@ -420,6 +428,225 @@ public void onError(Throwable t) {
         bodyContent.subscribe(s);
     }
 
+    @Override
+    public boolean isFull() {
+        return !bodyIsReadAsync;
+    }
+
+    @Override
+    public ByteBuffer<?> contents() {
+        if (bodyIsReadAsync) {
+            LOG.debug("Body is read asynchronously, cannot get contents");
+            return null;
+        }
+        try {
+            this.servletByteBuffer = new ServletByteBuffer<>(getInputStream().readAllBytes());
+            return servletByteBuffer;
+        } catch (IOException e) {
+            throw new IllegalStateException("Error getting all body contents", e);
+        }
+    }
+
+    @Override
+    public ExecutionFlow<ByteBuffer<?>> bufferContents() {
+        ByteBuffer<?> contents = contents();
+        if (contents == null) {
+            throw new IllegalStateException("Cannot buffer contents asynchronously");
```

Comment:
i changed my mind on this, please return null in this case (null, not ExecutionFlow.just(null)). I will adjust the supermethod in core accordingly.

Reply:
```suggestion
            return null;
```

Reply:
https://github.com/micronaut-projects/micronaut-servlet/pull/471/commits/a54657e018874a4a4a178a9b3181e61003bdff57